test-data/unit/check-abstract.test

```diff
@@ -157,6 +157,43 @@ class B(A): pass
 B()# E: Cannot instantiate abstract class 'B' with abstract attributes 'f' and 'g'
 [out]
 
+[case testInstantiationAbstractsInType]
+# flags: --python-version 3.6
```

Comment:
Since recently ``--python-version`` flag is not needed. Also I split the test into three smaller and added a test case for ``@classmethod``.